pkgs/development/tools/rust/cargo-workspaces/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, openssl
+, pkg-config
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-workspaces";
+  version = "0.2.34";
+
+  src = fetchFromGitHub {
+    owner = "pksunkara";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-7oJ/jjrv2XbOyLKUqp7EMFGR+czNSXCpWI8/JX5kmH8=";
+  };
+
+  cargoSha256 = "sha256-vZyodSxQd89dyaw4lr8vsbNjztXhFVgD0FcBP8KdPss=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  sourceRoot = "source";
+  cargoRoot = pname;
+  buildAndTestSubdir = pname;
```

Comment:
This repo is a little bit strange in that the Cargo.lock does not live at the root. Also, fixtures used by tests live in the root of the repo which caused some problems running tests. Setting `sourceRoot` and `cargoRoot` seemed to fix the issue for the patch phase but `buildAndTestSubdir` was also required for the build phase and check phases.